src/main/java/seedu/duke/commands/AddCommand.java

```diff
@@ -38,28 +39,32 @@ private Currency getCurrency (String currencyString) {
     }
 
     private void processCommand () throws InvalidAddCommandException,
-            InvalidAmountToAddException {
+            InvalidAmountToAddException, AmountTooPreciseException {
         String[] words = super.input.split(" ", 4);
         // Format: [Command, CURRENCY, AMOUNT, DESCRIPTION]
         boolean isValidCommand = words.length >= 3;
         if (!isValidCommand) {
             throw new InvalidAddCommandException();
         }
-        this.currency = getCurrency(words[1]);
+        currency = getCurrency(words[1]);
 
-        this.amount = new BigDecimal(words[2]);
-        if (this.amount.compareTo(BigDecimal.valueOf(0.01)) < 0) {
+        amount = new BigDecimal(words[2]);
+        if (amount.compareTo(BigDecimal.valueOf(0.01)) < 0) {
             throw new InvalidAmountToAddException();
         }
+        if (getNumberOfDecimalPlaces(amount) > 2) {
+            throw new AmountTooPreciseException();
+        }
```

Comment:
Great that you are adding this, I also added similar checks in show-rate. When this is merged I can standardize my checks to use AmountTooPrecise exceptions.